src/rust/engine/src/python.rs

```diff
@@ -391,13 +381,26 @@ impl From<PyObject> for Value {
 
 impl<'py, T> From<&Bound<'py, T>> for Value {
     fn from(value: &Bound<'py, T>) -> Self {
-        Value::new(value.clone().into_py(value.py()))
+        Value::new(
+            value
+                .clone()
+                .into_pyobject(value.py())
+                .unwrap()
```

Comment:
I'm not sure I like this `.unwrap()` even though a `Bound` should be convertible infallibly. Will try `.into_bound().into_any().unbind()` here instead.